### PR TITLE
Expanded init explanation when using nullability

### PR DESCRIPTION
**Pull request by EnessenE**
## Summary

I expanded the init explanation when using nullability as that wasn't mentioned but the behavior may be interesting for developers

It helps a bit with #27738

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/language-reference/keywords/init.md](https://github.com/dotnet/docs/blob/727b1e0cf825b52c17155d3e3f7346536b5e4894/docs/csharp/language-reference/keywords/init.md) | [init (C# Reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/init?branch=pr-en-us-35872) |

### CODE REVIEW

**Comment by EnessenE** [comment]
@dotnet-policy-service agree

File: docs/csharp/language-reference/keywords/init.md [hunk 1]
```diff
@@ -28,6 +28,9 @@ var john = new Person_InitExample
 
 john.YearOfBirth = 1926; //Not allowed, as its value can only be set once in the constructor
 ```
+Do note, that this behavior does not change when nullablity is used. `init` properties do not force the values to be initialized. What `init` does is that, if the property is to be initialized, it must be done at construction time in an object initializer (or in the constructor if you have one). So the behavior is the same for this example:
```
Comment (BillWagner) [inline docs/csharp/language-reference/keywords/init.md:31]: Two suggestions here:  First, add a blank line for markdown lint. Second, I made some clarifying comments on the text:

```suggestion

An `init` accessor doesn't force callers to set the property. Instead, it allows an object initializer to set the initial value while prohibiting later modification. You can add the `required` modifier to force callers to set a property. The following example shows the same behavior:
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
Hi @EnessenE 

I agree with adding clarification here. I've made some suggestions to fix build warnings, and to update the text. Let me know what you think, and then this is ready to merge.

File: docs/csharp/language-reference/keywords/init.md (hunk 1 above)
Comment (EnessenE) [reply docs/csharp/language-reference/keywords/init.md:31]: Agreed, my wording is a bit... direct. I like your wording alot better.

**Comment by EnessenE** [comment]
Agreed and all done!

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
closing and reopening for a fresh build (build warnings weren't introduced in this PR, and have since been fixed.)

**Pull request merged**